locust/cli.py

```diff
@@ -41,6 +46,9 @@ def main():
 
     results_string = render.run(parse_result, args.format, args.github)
 
+    if args.metadata and args.format == "json":
```

Comment:
@kompotkot, 

We should provide a consistent interface across all formats. A better way to do this is to modify `locust/render.py`.

1. Add `--metadata` argument in `populate_argument_parser`. (Give it a type of `json.loads` and a default value of `{}`.)

2. Add an `enrich_with_metadata function` which accepts `args.metadata` and enriches the `results` object (see [`enrich_with_refs`](https://github.com/bugout-dev/locust/blob/main/locust/render.py#L258) for an example of this).